src/components/website-management/website-form-components/customization/customization-components/BannerTemplate.js

```diff
@@ -0,0 +1,160 @@
+    templates?.length > 0
+      ? templates.map((template) => ({
+          id: template.templateId,
+          value: template.templateRegulation,
+        }))
+      : [];
+
+  const choosenTemplate = (e) => {
+    const template = templates.find(
+      (t) => t.templateRegulation === e.target.value
+    );
+    setSelectedTemplate(template);
+  };
+
+  return (
+    <div className={"TemplateView"}>
+      <Box
+        sx={{
+          backgroundColor: "#fefefe",
+          width: "450px",
+          height: "650px",
+          align: "center",
+          margin: 3,
+          paddingTop: 5,
+          paddingLeft: 5,
+          borderRadius: 5,
+          boxShadow: "4px 4px 20px 4px rgba(0, 0, 0, 0.25)",
+          mb: 8,
+        }}
+      >
+        <Typography
+          variant={"h6"}
+          sx={{ color: "#00A5FF", pb: { lg: 3, md: 3, sm: 3, xs: 3 } }}
+          fontWeight={"Bold"}
+        >
+          Banner Template
+        </Typography>
+        <div
+          style={{
+            paddingTop: 10,
+            paddingBottom: 10,
+            textAlign: "right",
+            paddingRight: 45,
+          }}
+        >
+          <CustSelectBox
+            name={"Default template"}
+            items={regulationValues}
+            width={120}
+            fun={choosenTemplate}
+            selectedTemplate={selectedTemplate}
+          />
+        </div>
+        <div style={{ paddingTop: 10, paddingBottom: 10 }}>
+          <TextField
+            sx={{ width: "40ch" }}
+            id={String(selectedTemplate?.templateId) || ""}
+            label="Template Name"
+            variant="outlined"
+            value={selectedTemplate?.templateName || ""}
+            InputLabelProps={{ shrink: true }}
+          />
+        </div>
+        <div style={{ paddingTop: 10, paddingBottom: 10 }}>
+          <TextField
+            sx={{ width: "40ch" }}
+            id={String(selectedTemplate?.templateId) || ""}
+            label="Privacy Regulation Type"
+            variant="outlined"
+            value={selectedTemplate?.templateRegulation || ""}
+            InputLabelProps={{ shrink: true }}
+          />
+        </div>
+        <div style={{ paddingTop: 10, paddingBottom: 10 }}>
+          <TextField
+            sx={{ width: "40ch" }}
+            id={String(selectedTemplate?.templateId) || ""}
+            label="Cookie policy link"
+            variant="outlined"
+            value={selectedTemplate?.templatePrivacyPolicyLink || ""}
+            InputLabelProps={{ shrink: true }}
+          />
+        </div>
+        <div style={{ paddingTop: 10, paddingBottom: 10 }}>
+          <TextField
+            sx={{ width: "40ch" }}
+            id={String(selectedTemplate?.templateId) || ""}
+            label="Content"
+            variant="outlined"
+            value={selectedTemplate?.templateContent || ""}
+            multiline={true}
+            InputLabelProps={{ shrink: true }}
+          />
+        </div>
```

Comment:
We can iterate through a value array and dynamically add these elements instead of hardcoding each element.

Eg:

```js
const values = [template.name,template.regulation,template.link];

// in jsx
{ values.map(value => 
     <div>
        <TextField 
              ...other props
              value={value} />
     </div>
)}
```

Reply:
noted.